### PR TITLE
Pepperbox balance fix and json cleanup for paper ammo

### PR DESCRIPTION
**Pull request by Treah**
This pull will update the pepperbox pistol to use ammo more suited to its size and reduce its damage output. This resolves issue #18483

Also updated the json files to separate the guns and ammo for paper cartridge guns. I omitted the flintlock stuff as its already in its own json file and its possible it may be reworked to no longer use paper carts. Below is a screen cap of the updated damage outputs for the pepperbox. 

image

### CODE REVIEW

**Comment by DangerNoodle** [comment]
Seems simple and acceptable overall. JSON placement I have no idea about. One the one hand, currently each ammo type seems to get its own separate gun and ammo entry. On the other hand, this pattern might cause a small degree of file bloat if taken to its logical conclusion.

File: data/json/items/ammo/paper.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
+[
```
Comment (mugling) [inline data/json/items/ammo/paper.json]: This file needs to be named `36paper.json` and linted

**Comment by mugling** [comment]
> JSON placement I have no idea about

Please follow the existing convention of one `ammo_type` per file. Arguably the paper ammo types could be combined?

**Comment by Treah** [comment]
Ahh so you would like them in 44paper.json and 36paper.json or just 44.json and 36.json? Also should each of the guns then be moved into similar files in guns? Also Im not sure about combining the types as they have different damage stats and it would all anything that can use paper ammo to use any caliber. Unless of course I am misunderstanding.

**Comment by DangerNoodle** [comment]
> Arguably the paper ammo types could be combined?

Combining as in making a .36 caliber revolver and a .44 caliber one use the same ammo type? That would largely eliminate the differences between the two revolvers, and not really add enough benefit to warrant it.

**Comment by mugling** [comment]
They aren't good items as it stands as they are entirely eclipsed by the other revolvers.

One possible niche would be to combine/reduce them to a single definition and then make the ammo craftable from percussion caps plus lead and blackpowder. This makes them useful where the player lacks casings or gunpowder.

**Comment by Treah** [comment]
Hmmm mugling I like that idea a lot more then spreading out the ammo types into yet more files and agree the weapons are not very good as it stands. Since the scope of this PR is to fix the damage on the pepperbox I can close this PR and then make a new pr to eliminate all 3 different types of paper ammo, combine them into damage levels somewhere in between and make them craftable via your suggestion.

**Comment by DangerNoodle** [comment]
That is a woefully more complicated idea that risks causing more problems that it will solve.

Combining all 3 is also a bad idea. Flintlock ammo does not use percussion caps, unlike the cap-and-ball (as the name implies) ammo.

**Comment by Treah** [comment]
Disregard the flintlock as it uses a flintlock ammo type not paper ammo. So i would leave that one alone as your correct danger it does not use percussion caps.

**Comment by mugling** [comment]
Yes, leave flintock as it is as @DangerNoodle is already working that in to a suitable niche and there are meaningful differences there.

**Comment by DangerNoodle** [comment]
On the one hand, I am uncertain about this.

On the other hand, flintlock cartridges are already vague in caliber. The pistol, carbine, and rifle could all be stand-ins 

However, the revolvers stand out by being specific models 

It might be sensible to balance them a bit closer to their modern equivalents rather than balancing via genericness. Depending on the load, cap-and-ball rounds can get terminal ballistics that are acceptable for self-defense, though they tend to suffer in various other ways (effectiveness against armor versus FMJ, expansion versus hollowpoints).

Those are more likely if using modern blackpowder substitutes specifically designed for reproductions, genuine antiques might not be able to handle higher pressures that would improve performance. Handwaving any cap-and-ball revolver as being vaguely either an antique or a reproduction is likely more acceptable to gun enthusiasts than handwaving both calibers of cap-and-ball as being interchangeable.

**Comment by mugling** [comment]
If we are keeping two variants then one could use ammo crafted with large pistol primers, and one with small pistol primers?

**Comment by DangerNoodle** [comment]
That was another change I was considering. I had yet to look up what primers are used in the recipes.

Do we even use primers to craft those at the moment?

**Comment by mugling** [comment]
> Do we even use primers to craft those at the moment?

No. We still need to address the issue as to whether primers should be craftable. My gut feeling from a gameplay POV is yes but with a better recipe than current

**Comment by DangerNoodle** [comment]
I recall the use of strike-anywhere matches as a source of makeshift primer for some out-of-production cartridges. Matches in-game are implied to be strike-on-box though, at least I assumed.

As for other uses, I am unsure. Leaving them craftable in some way seems fine, we just need a sensible ingredient.

**Comment by Treah** [comment]
Gonna rework this PR to only address the pepperbox gun and not the larger issues with paper cartridges. We can revisit that at some point. Give me a few to make the necessary changes.

**Comment by DangerNoodle** [comment]
No worries. For now we can always split them up into 36paper and 44paper, then rebalance or otherwise decide how to handle them later on. I might incorporate a rebalance of the cap-and-ball revolver ammo in a later pull request, as I had more general "older firearm balancing" planned after I am done with my current pull requests.

**Comment by Treah** [comment]
> No worries. For now we can always split them up into 36paper and 44paper gun/recipe files
This is sorta what I did, I split them into different json files for the paper ammo and gun types. The recipes for ammo are all jumbled into a huge file that contains the recipe for these. I didn't touch them or modify them in any way, however their recipe is a little unrealistic as it allows you to use plastic bags and not need any glue.

**Comment by DangerNoodle** [comment]
Excellent then. As for the recipes, I assumed that the glue was abstracted out.

Using plastic definitely sounds like the recipe designer is implying the use of torn-open cartridges, rather than nitrated paper. When I get around to that rebalancing pull request, I will double-check the reload times for the revolvers, then decide where to go from there if the implications do not match up.

**Comment by Treah** [comment]
Yeah I think they were def thinking that these were the simple baggie type of ammo rather than the nitrated carts type. When you redo them it might be a good use for bone glue since it only has 18 or so recipes associated with it if we keep the cartridges and not go to a different system. 

I was also tempted to reduce the gun down to only 3 shots rather than 6. I can't find a reference to pepperbox ever having that many cylinders. The 3 was may more prevalent at least from online research as I am not a gun enthusiast.

**Comment by DangerNoodle** [comment]
Possibly, though it depends on how needed this addition really is. While wikipedia does mention that as one of several capacities (including six), I would feel more confident about recommending a number if I could find other sources citing specific models.

**Comment by DangerNoodle** [comment]
Also, regarding future changes to cap-and-ball ammunition, it seems like it might be acceptable to increase their damage by a small amount (+2 for .36, +1 for .44), but I am unsure what further changes will make them more useful.

One added change might be to reduce dispersion. The flintlock ammo adds 90 to to dispersion, whereas both cap-and-ball rounds have 120. Reducing them to a lower value would balance them against both modern ammo and the flintlock paper cartridges.

Conversely, the revolvers have reload times of 130 (M1861) and 150 (M1860). If these are implied to not use nitrated paper, they could be increased with the standard revolver reloading time (70) in mind. For comparison, the flintlock pistol has a reload time of 600.

**Comment by DangerNoodle** [comment]
If desired, I could incorporate planned ammo rebalancing and this fix in the same pull request. And lintfix it (though there is http://dev.narc.ro/cataclysm/format.html for automated formatting).

**Comment by DangerNoodle** [comment]
Also @mugling, apologies for this being a bit off-topic, but I did find a video relevant to the subject of realistically making new primers: https://www.youtube.com/watch?v=t_7LWCFH5Gc

Even better than expected, as this means the issue of "matches in-game are likely strike-on-box" goes out the window.

**Comment by Treah** [comment]
>If desired, I could incorporate planned ammo rebalancing and this fix in the same pull request. And lintfix it 

@DangerNoodle - I lint fixed the PR but I don't mind if you merge this with your PR. I can close if it you do.

**Comment by mugling** [comment]
> Also @mugling, apologies for this being a bit off-topic, but I did find a video relevant to the subject of realistically making new primers: https://www.youtube.com/watch?v=t_7LWCFH5Gc
> 
> Even better than expected, as this means the issue of "matches in-game are likely strike-on-box" goes out the window.

Looks good. I might write something up using that

File: data/json/items/gun/36paper.json [hunk 2]
```diff
@@ -1,29 +1,31 @@
+Cataclysm web linting tool
```
Comment (mugling) [inline data/json/items/gun/36paper.json]: Well it's nice to see it getting some use but I think you got a bit carried away with the pasting there

Comment (Treah) [inline data/json/items/gun/36paper.json]: Whoops thought i fixed that.... This happend when my VM was dying and copy and paste were getting weird

**Comment by Treah** [comment]
Its all good if you added this as part of your PR DangerNoodle ill close this one.

**Comment by Treah** [comment]
Redundant as danger has incorporated this into another PR

**Comment by DangerNoodle** [comment]
Apologies for the best. Hopefully including the ammo rebalancing does not significantly delay things though.

**Pull request closed without merging**